quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/services/health/KeycloakReadyHealthCheck.java

```diff
@@ -31,15 +35,34 @@
 
 /**
  * Keycloak Healthcheck Readiness Probe.
- *
+ * <p>
  * Performs a hybrid between the passive and the active mode. If there are no healthy connections in the pool,
- * it invokes the standard <code>DataSourceHealthCheck</code> that creates a new connection and checks if its valid.
+ * it invokes the standard <code>DataSourceHealthCheck</code> that creates a new connection and checks if it's valid.
+ * <p>
+ * While the check for healthy connections is non-blocking, the standard check is blocking, so it needs to be wrapped.
  *
  * @see <a href="https://github.com/keycloak/keycloak-community/pull/55">Healthcheck API Design</a>
  */
 @Readiness
 @ApplicationScoped
-public class KeycloakReadyHealthCheck extends DataSourceHealthCheck {
+public class KeycloakReadyHealthCheck implements AsyncHealthCheck {
+
+    /** As the DataSourceHealthCheck doesn't exist as an application scoped bean,
+    * create our own instance here which exposes the <code>init()</code> call for the delegate. */
```

Comment:
@ahus1 Does not exist as an application-scoped bean? For Quarkus 3.2.5, obtaining the instance through the CDI should be possible. Can you elaborate on it?

https://github.com/quarkusio/quarkus/blob/3.2/extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/health/DataSourceHealthCheck.java#L25

Reply:
Keycloak has explicitly disabled it, see:

https://github.com/keycloak/keycloak/blob/3c00286e6cd51cba843b5e08aff07ddc83e28a74/quarkus/runtime/src/main/resources/application.properties#L10

I assume it has been disabled as this allowed overwriting the health check with a Keycloak's custom database check.